### PR TITLE
Update Sleuth:  complex experimental design mode 

### PR DESCRIPTION
**Pull request by gallardoalba**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/sleuth/sleuth.xml [hunk 1]
```diff
@@ -46,17 +61,30 @@
...
-        <repeat name="rep_factorLevel" title="Factor level" min="2" default="2">
-            <param name="factorLevel" type="text" value="FactorLevel" label="Specify a factor level, typical values could be 'tumor', 'normal', 'treated' or 'control'"
-                help="Only letters, numbers and underscores will be retained in this field">
-                <sanitizer>
-                    <valid initial="string.letters,string.digits"><add value="_" /></valid>
-                </sanitizer>
+        <conditional name="experiment_design">
+            <param name="selector" type="select" label="Experiment design" help="If you have multiple experimental conditions, you should use propably the complex design mode. In the help section you can find more information.">
+                <option value="single">Simple design mode (one experimental factor)</option>
+                <option value="complex">Complex design mode (two experimental factorse)</option>
```
Comment (bgruening) [inline tools/sleuth/sleuth.xml:70]: here is a typo `factorse`. Isn't that mode unrestricted with the amount of factors?

File: tools/sleuth/sleuth.xml [hunk 2]
```diff
@@ -123,6 +174,24 @@ pseudo-alignment. An important feature of kallisto is that it outputs bootstraps
...
+        ::
+        
+            data_filename	condition	sample
+            finename1	condition1	replicate1
+            filename2	condition1  replicate2
+            filename3	condition2	replicate1
+            filename4	condition2	replicate2
+
+
+The tabular file requires to have the same column names as the example (data_file, condition, sample). The data file column correspond to the input filenames.
```
Comment (gallardoalba) [inline tools/sleuth/sleuth.xml:192]:
```suggestion
The tabular file **requires to have the same column names** as the example (data_file, condition, sample). The data file column correspond to the input filenames. 
```

Comment (bgruening) [inline tools/sleuth/sleuth.xml]: What is a filename here? That is the Galaxy dataset name?

Comment (gallardoalba) [reply tools/sleuth/sleuth.xml]: Filename correspond to the original files used in the workflow. I think it is important in order to track the files correctly.  I'll include that information.

**Comment by gallardoalba** [comment]
I'll create also a small tutorial in order to explain how to make use of the complex design mode.

File: tools/sleuth/sleuth.xml [hunk 3]
```diff
@@ -123,6 +174,24 @@ pseudo-alignment. An important feature of kallisto is that it outputs bootstraps
...
+
+**Experimental design tabular input for complex experimental designs**
+
+The experimental design input should have this format:
+
+        ::
+        
+            data_filename	condition	sample
+            finename_01.fastq.gz	condition1	replicate1
+            filename_02.fastq.gz	condition1  replicate2
```
Comment (gallardoalba) [inline tools/sleuth/sleuth.xml:187]:
```suggestion
            finename_01.fastq.gz	condition1	replicate1
            filename_02.fastq.gz	condition1	replicate2
```

File: tools/sleuth/sleuth.R [hunk 4]
```diff
@@ -26,39 +26,67 @@ loc <- Sys.setlocale("LC_MESSAGES", "en_US.UTF-8")
...
+               sep = "\t")
+  paths <- c()
+  for (x in s2c$data_filename) {
+    paths <- c(paths, paste("./kallisto_outputs/", x, sep = ""))
+  }
+  for (f in paths) {
+    file.rename(f, gsub(".fastq.*", "", f))
+        file.rename(f, paste(gsub(".fastq.*", "", f),".h5",sep=""))
+  }
+  s2c$path <- paste(gsub(".fastq.*", ".h5", paths),".h5",sep="")
```
Comment (gallardoalba) [inline tools/sleuth/sleuth.R:62]:
```suggestion
  paths <- c()
  for (x in s2c$data_filename) {
    paths <- c(paths, paste("./kallisto_outputs/",  x, sep = ""))
  }
  for (f in paths) {
    file.rename(f, gsub(".fastq.*", "", f))
        file.rename(f, paste(gsub(".fastq.*", "", f), ".h5",sep = ""))
  }
  s2c$path <- paste(gsub(".fastq.*", ".h5", paths), ".h5",sep = "")
```

File: tools/sleuth/sleuth.R [hunk 5]
```diff
@@ -26,39 +26,67 @@ loc <- Sys.setlocale("LC_MESSAGES", "en_US.UTF-8")
...
+               sep = "\t")
+  paths <- c()
+  for (x in s2c$data_filename) {
+    paths <- c(paths, paste("./kallisto_outputs/",  x, sep = ""))
+  }
+  for (f in paths) {
+    file.rename(f, gsub(".fastq.*", "", f))
+        file.rename(f, paste(gsub(".fastq.*", "", f), ".h5",sep = ""))
+  }
+  s2c$path <- paste(gsub(".fastq.*", ".h5", paths), ".h5",sep = "")
```
Comment (gallardoalba) [inline tools/sleuth/sleuth.R:62]:
```suggestion
        file.rename(f, paste(gsub(".fastq.*", "", f), ".h5", sep = ""))
  }
  s2c$path <- paste(gsub(".fastq.*", ".h5", paths), ".h5", sep = "")
```

**Review by bgruening — APPROVED** [review]

**Pull request merged**